FAO Publications.js

```diff
@@ -445,27 +441,27 @@ var testCases = [
 	},
 	{
 		"type": "web",
-		"url": "http://www.fao.org/publications/card/en/c/4ca616af-0a4a-4232-bd3b-681b67471857/",
+		"url": "http://www.fao.org/documents/card/en/c/ca7988en/",
 		"items": [
 			{
 				"itemType": "book",
-				"title": "Save Food for a Better Climate: Converting the food loss and waste challenge into climate action",
+				"title": "FAO publications catalogue 2020: March",
 				"creators": [
 					{
 						"lastName": "FAO",
-						"fieldMode": true
+						"creatorType": "author"
 					}
 				],
-				"date": "2017",
-				"ISBN": "9789251099896",
-				"abstractNote": "This paper aims to inform on the interrelationship between food loss and waste and climate change. In this context, the paper highlights the related impacts, and outlines the recent global frameworks adopted by the international community, and how they have been translated into national priorities and targets. Climate technology options are explored, along with the challenges and opportunities related to financing needs. Finally, this paper will identify ways and enabling factors to reduce food loss and waste as part of the collective effort to enhance ambition for climate action while simultaneously delivering the other objectives of the sustainable development agenda.",
+				"date": "2020",
+				"ISBN": "9789251322550",
+				"abstractNote": "This catalogue aims to improve the dissemination and outreach of FAO’s knowledge products and overall publishing programme. By providing information on its key publications in every area of FAO’s work, and catering to a range of audiences, it thereby contributes to all organizational outcomes.\n\nFrom statistical analysis to specialized manuals to children’s books, FAO publications cater to a diverse range of audiences. This catalogue presents a selection of FAO’s main publications, produced in 2020 or earlier, ranging from its global reports and general interest publications to numerous specialized titles. In addition to the major themes of agriculture, forestry and fisheries, it also includes thematic sections on climate change, economic and social development, and food safety and nutrition.",
 				"language": "en",
 				"libraryCatalog": "FAO Publications",
-				"numPages": "38",
+				"numPages": "114",
 				"place": "Rome, Italy",
 				"publisher": "FAO",
-				"shortTitle": "Save Food for a Better Climate",
-				"url": "www.fao.org/3/a-i8000e.pdf",
+				"shortTitle": "FAO publications catalogue 2020",
+				"url": "www.fao.org/3/ca7988en/CA7988EN.pdf",
```

Comment:
Yes this is a tricky issue. I'd stick to the PDF URL and without "http://" because the style is set this way. This can be changed but that needs discussion with the editors of FAO. And the URLs of abstract pages is too long.

However, DOIs have been assigned to publications since 2020, which is why older publications, like "Save Food for a Better Clilmate", doesn't have a DOI. I don't know they'll be assigned retrospectively but I guess it's unlikely to happen. I have adjusted the code so that if the publication has a DOI, the DOI url is used instead of the PDF URL. This is reflected in the test cases [](http://www.fao.org/documents/card/en/c/ca8751en/) and [](http://www.fao.org/documents/card/en/c/ca7988en/). I can confirm that the Extra field does contain DOI for books if assigned.

Reply:
> I'd stick to the PDF URL and without "http://" because the style is set this way.

Well, no — this is the Zotero URL field. It needs to be a full URL with a scheme. The data isn't tied to any given citation style, so what FAO editors want in the FAO style is irrelevant for what Zotero saves to the field, which can be used for other uses and other styles. (But they also shouldn't be dictating the use of URLs without schemes — they don't get to just make up what a URL looks like. It's technical malpractice that they're not using HTTPS for their site anyway, and while there's a way to register domains with browsers so that they default to loading via HTTPS, that's not universally done (e.g., by worldbank.org, which they give as an example), so FAO shouldn't be encouraging insecure initial connections.)

> I have adjusted the code so that if the publication has a DOI, the DOI url is used instead of the PDF URL.

There's no point in using a DOI URL in the URL field — the DOI can already be included and linked by styles, with a fallback to the URL field. For consistency with their own DOIs and general Zotero practices, it should really be the abstract page. (The length of their URLs isn't our concern.)

Reply:
Thanks. Your reasons are valid. I've adjusted the code accordingly.